app/mm-redux/actions/integrations.ts

```diff
@@ -399,3 +409,62 @@ export function submitInteractiveDialog(submission: DialogSubmission): ActionFun
         return {data};
     };
 }
+
+export function handleGotoLocation(href: string, intl: any): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState();
+        const config = getConfig(state);
+
+        const {url, match} = await DraftUtils.getURLAndMatch(href, getCurrentUrl(state), config.SiteURL);
+
+        if (match) {
+            switch (match.type) {
+            case DeepLinkTypes.CHANNEL:
+                dispatch(handleSelectChannelByName(match.channelName, match.teamName, () => DraftUtils.errorBadChannel(intl)));
+                break;
+            case DeepLinkTypes.PERMALINK:
+                dispatch(loadChannelsByTeamName(match.teamName, () => DraftUtils.errorPermalinkBadTeam(intl)));
+                if (match.postId) {
+                    dispatch(showPermalinkView(match.postId));
+                }
+                break;
+            case DeepLinkTypes.DMCHANNEL: {
+                const {data} = await dispatch(getUserByUsername(match.userName || ''));
+                if (!data) {
+                    DraftUtils.errorBadUser(intl);
+                    return {data: false};
+                }
+                dispatch(makeDirectChannel(data.id));
+                break;
+            }
+            case DeepLinkTypes.GROUPCHANNEL:
+                dispatch(handleSelectChannel(match.id));
+                break;
+            }
+        } else {
+            DraftUtils.tryOpenURL(url);
+        }
+        return {data: true};
+    };
+}
+
+export function showPermalinkView(postId: string, error = ''): ActionFunc {
```

Comment:
this is already defined here https://github.com/mattermost/mattermost-mobile/blob/master/app/actions/views/permalink.ts#L16